### PR TITLE
Omar/hns target delete

### PR DESCRIPTION
**Pull request by OmarKhatib158**
Added support for deleting files and folders in blobfs target.
Fixed a bug when metadata properties header is empty

### CODE REVIEW

File: cmd/syncProcessor.go [hunk 1]
```diff
@@ -432,6 +433,15 @@ func (b *remoteResourceDeleter) deleteFolderRecursively(object StoredObject) err
 			fmt.Printf("Object [%s] Deletion failed with error: %v", object.relativePath, err)
 			return err
 		}
+	case common.ELocation.BlobFS():
+		blobUrlParts := azbfs.NewBfsURLParts(*b.rootURL)
+		blobUrlParts.DirectoryOrFilePath = path.Join(blobUrlParts.DirectoryOrFilePath, object.relativePath)
+		dirUrl := azbfs.NewFileSystemURL(blobUrlParts.URL(), b.p)
+		_, err := dirUrl.Delete(b.ctx)
```
Comment (afekson) [inline cmd/syncProcessor.go:440]: If it is a directory, can we do a recursive delete?

File: cmd/syncProcessor.go [hunk 2]
```diff
@@ -476,6 +486,11 @@ func (b *remoteResourceDeleter) delete(object StoredObject) error {
 					}
 				}
 			}
+		case common.ELocation.BlobFS():
```
Comment (afekson) [inline cmd/syncProcessor.go:489]: Here and in line 438 there is the same code. Can we extract it to a common function?

File: azbfs/url_filesystem.go [hunk 3]
```diff
@@ -67,7 +67,7 @@ func (s FileSystemURL) Create(ctx context.Context) (*FilesystemCreateResponse, e
 // Delete marks the specified file system for deletion.
 // The file system and any files contained within it are later deleted during garbage collection.
 func (s FileSystemURL) Delete(ctx context.Context) (*FilesystemDeleteResponse, error) {
-	return s.fileSystemClient.Delete(ctx, s.name, nil, nil, nil, nil, nil)
+	return s.fileSystemClient.Delete(ctx, true, s.name, nil, nil, nil, nil, nil)
```
Comment (afekson) [inline azbfs/url_filesystem.go:70]: What is the usage of this flag?

File: cmd/syncProcessor.go [hunk 4]
```diff
@@ -432,6 +433,25 @@ func (b *remoteResourceDeleter) deleteFolderRecursively(object StoredObject) err
 			fmt.Printf("Object [%s] Deletion failed with error: %v", object.relativePath, err)
 			return err
 		}
+	case common.ELocation.BlobFS():
+		blobUrlParts := azbfs.NewBfsURLParts(*b.rootURL)
+		blobUrlParts.DirectoryOrFilePath = path.Join(blobUrlParts.DirectoryOrFilePath, object.relativePath)
+		dirUrl := azbfs.NewDirectoryURL(blobUrlParts.URL(), b.p)
+		marker := ""
+		for {
+			resp, err := dirUrl.Delete(b.ctx, &marker, true)
```
Comment (afekson) [inline cmd/syncProcessor.go:442]: Is it recursive ?

File: cmd/syncProcessor.go [hunk 5]
```diff
@@ -432,6 +433,25 @@ func (b *remoteResourceDeleter) deleteFolderRecursively(object StoredObject) err
...
+			if err != nil {
+				return err
```
Comment (shmuelberezin) [inline cmd/syncProcessor.go:444]: In the other 2 cases, there is a log print. Maybe worth adding also here?
fmt.Printf("Folder [%s] Delete failed with error: %v", object.relativePath, err)

File: cmd/syncProcessor.go (hunk 4 above)
Comment (OmarKhatib158) [reply cmd/syncProcessor.go:446]: Yes, It would delete sub directories and files

**Review by shmuelberezin — APPROVED** [review]

**Pull request merged**